lib/public/TextProcessing/IProvider2.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Marcel Klehr <mklehr@gmx.net>
+ *
+ * @author Marcel Klehr <mklehr@gmx.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+namespace OCP\TextProcessing;
+
+/**
+ * This interface supersedes IProvider. It allows the system to learn
+ * the provider's expected runtime and lets the provider know which user is running a task
+ * @since 28.0.0
+ * @template T of ITaskType
+ * @template-extends IProvider<T>
+ */
+interface IProvider2 extends IProvider {
```

Comment:
```suggestion
interface IProviderWithExpectedRuntime extends IProvider {
```
Or something descriptive like that

Reply:
IProviderWithExpectedRuntimeAndUserId :thinking: ?

Reply:
Personally I'm in favor of IProvider2 because it supersedes IProvider and is not just an addendum

Reply:
But will you ever be able to discard the previous one? For user backends we use addendums, and here you could split into `IExpectedRuntimeAwareProvider` and `IUserAwareProvider`.
But yeah fine with `IProvider2` if you know what you’re doing.

Reply:
I decided to split IProvider2 into `IProviderWithUserId` and `IProviderWithExpectedRuntime`